src/main/java/com/atlassian/db/replica/api/KeyBasedThrottledCache.java

```diff
@@ -0,0 +1,43 @@
+package com.atlassian.db.replica.api;
+
+import com.atlassian.db.replica.spi.SuppliedCache;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+/**
+ * Every read can potentially refresh the cache, however it can be loaded only by one thread at the time.
+ * Other threads instead of waiting are getting last known value.
+ */
+public class KeyBasedThrottledCache implements SuppliedCache<Long> {
+    private static final ConcurrentHashMap<String, Long> lastValues = new ConcurrentHashMap<>();
```

Comment:
The `static` makes testing and GC harder.

If the keying aspect is decoupled or removed, the locking aspect will be easier to test in integration tests.